idseq_pipeline/commands/common.py

```diff
@@ -171,29 +172,49 @@ def get_remaining_reads_from_stats():
     return (item for item in STATS if item.get("task") == "run_gsnapl_remotely").next().get("reads_before")
 
 def run_and_log(logparams, target_outputs, lazy_run, func_name, *args):
+    global OUTPUT_VERSIONS
     global LOGGER
     LOGGER = logging.getLogger()
     LOGGER.info("========== %s ==========" % logparams.get("title"))
+
     # copy log file -- start
     LOGGER.handlers[0].flush()
     execute_command("aws s3 cp %s %s/;" % (LOGGER.handlers[0].baseFilename, logparams["sample_s3_output_path"]))
+
+    # record version of any reference index used
+    version_file_s3 = logparams.get("version_file_s3")
+    if version_file_s3:
+        version_json = execute_command_with_output("aws s3 cp %s -" % version_file_s3)
+        OUTPUT_VERSIONS.append(version_json)
+
+    # upload version output
+    output_version_file = logparams.get("output_version_file")
+    if output_version_file:
+        with open(output_version_file, 'wb') as f:
+            json.dump(OUTPUT_VERSIONS, f)
+        execute_command("aws s3 cp %s %s/;" % (output_version_file, logparams["sample_s3_output_path"]))
```

Comment:
if you wanted to stream into aws s3 cp, you could do it with python subprocess here... using the PIPE args

Reply:
Ah! That's cool!